tools/testrunner.py

```diff
@@ -273,32 +255,42 @@ def run_test(self, testfile, timeout):
         return exitcode, stdout, runtime
 
     def skip_test(self, test):
-        skip_list = test.get("skip", [])
+        skip_list = set(test.get("skip", []))
 
         # Skip by the `skip` attribute in testsets.json file.
         for i in ["all", Platform().os(), self.stability]:
             if i in skip_list:
                 return True
 
-        name_parts = test["name"][0:-3].split('_')
+        required_modules = set(test.get("required-modules", []))
+        required_features = set(test.get("required-features", []))
 
-        # Test filename does not start with 'test_' so we'll just
-        # assume we support it.
-        if name_parts[0] != 'test':
-            return False
+        unsupported_modules = required_modules - self.builtins
+        unsupported_features = required_features - self.features
+        skipped_modules = required_modules.intersection(skip_list)
 
-        tested_module = name_parts[1]
+        # Skip the test if the tested module requires a module
+        # which is not compiled into the binary
+        if unsupported_modules:
+            test["reason"] = "Required module(s) \""
+            test["reason"] += '\", \"'.join(sorted(unsupported_modules))
+            test["reason"] += "\" unsupported by iotjs build"
```

Comment:
What if we use this output format?: `Required module(s) unsupported by build: tls, kkk,`

Reply:
Yes, maybe that will be a better format